contracts/BaseStrategy.sol

```diff
@@ -732,38 +749,34 @@ abstract contract BaseStrategy {
      * @return `true` if `harvest()` should be called, `false` otherwise.
      */
     function harvestTrigger(uint256 callCostInWei) public view virtual returns (bool) {
-        uint256 callCost = ethToWant(callCostInWei);
-        StrategyParams memory params = vault.strategies(address(this));
+        // Should not trigger if strategy is not active (no assets or no debtRatio)
+        if (!isActive()) return false;
+
+        // check if the base fee gas price is higher than we allow. if it is, block harvests.
+        if (!isBaseFeeAcceptable()) return false;
 
-        // Should not trigger if Strategy is not activated
-        if (params.activation == 0) return false;
+        // trigger if we want to manually harvest, but only if our gas price is acceptable
+        if (forceHarvestTriggerOnce) return true;
 
         // Should not trigger if we haven't waited long enough since previous harvest
+        StrategyParams memory params = vault.strategies(address(this));
         if ((block.timestamp - params.lastReport) < minReportDelay) return false;
 
         // Should trigger if hasn't been called in a while
         if ((block.timestamp - params.lastReport) >= maxReportDelay) return true;
```

Comment:
Arent these checks back to back redundant? This would only matter if for some reason minReport is > maxReport which doesnt make sense.

Reply:
it would prevent harvesting the available credit if it hadn't been long enough yet, but in general I do agree that `minReportDelay` isn't super helpful in this setting. 

Maybe worth removing it here and allowing others to implement via `super.harvestTrigger`?

Reply:
That would probably be my thought. Seems like most of the time you would want credit available to cause a harvest even when under the minReport unless it was a strategy that has specific lockup times, which should probably get manually implemented in an override version anyway